client/src/constants/validation.js

```diff
@@ -173,12 +180,12 @@ export const EmployerFormSchema = yup.object().noUnknown('Unknown field for form
 });
 
 export const InterviewingFormSchema = yup.object().noUnknown('Unknown field in form').shape({
-  contactedDate: yup.string().required('Date of contact is required').test('is-date', 'Not a valid date', validateDateString),
+  contactedDate: yup.string().required('Date of contact is required').test('is-date', 'Not a valid date in the past', validatePastDateString),
 });
 
 export const HireFormSchema = yup.object().noUnknown('Unknown field in form').shape({
   startDate: yup.string().required('Start date is required').test('is-date', 'Not a valid date', validateDateString),
-  hiredDate: yup.string().required('Date hired is required').test('is-date', 'Not a valid date', validateDateString),
+  hiredDate: yup.string().required('Date hired is required').test('is-date', 'Not a valid date', validatePastDateString),
```

Comment:
Nitpick: Should we match the error messages?
```suggestion
  hiredDate: yup.string().required('Date hired is required').test('is-date', 'Not a valid date in the past', validatePastDateString),
```